ruby/activities/base_activity.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'temporalio/activity'
+
+module Activities
+  class BaseActivity < Temporalio::Activity::Definition
+    RETRY_POLICY = Temporalio::RetryPolicy.new(
+      initial_interval: 1,
+      backoff_coefficient: 1.2
+    )
+
+    private
+
+    def simulate_external_operation(ms)
+      sleep(ms / 1000.0)
+    end
+
+    def simulate_external_operation_with_error(ms, workflow_type, attempt)
+      simulate_external_operation(ms / attempt)
+      attempt < 5 ? workflow_type : 'NoError'
+    end
+
+    def logger
+      @logger ||= Temporalio::Activity::Context.current.logger
```

Comment:
This is a bit dangerous. If an activity is given to a worker already constructed, it reuses the same instance for _every_ activity attempt, but if it's given as a class to the worker, it is re-instantiated for _every_ attempt. Most activities I suspect may use the former because they have database clients or other dependencies they want to provide at activity creation time. But this kind of base class helper assumes the latter and this sample assumes the latter.

Reply:
@cretz Ah, this is my oversight. Fixed at here: https://github.com/temporal-sa/money-transfer-demo/pull/16/commits/b1767ab5746a89e3f380e1c07b2cf8986112f625